packages/rum/src/viewCollection.ts

```diff
@@ -41,13 +41,24 @@ export function startViewCollection(location: Location, lifeCycle: LifeCycle) {
   const startOrigin = 0
   let currentView = newView(lifeCycle, currentLocation, ViewLoadingType.INITIAL_LOAD, startOrigin)
 
+  function onRouteChange() {
+    currentLocation = { ...location }
+    currentView.triggerUpdate()
+    currentView.end()
+    currentView = newView(lifeCycle, currentLocation, ViewLoadingType.ROUTE_CHANGE)
+  }
+
   // Renew view on history changes
   trackHistory(() => {
     if (areDifferentViews(currentLocation, location)) {
-      currentLocation = { ...location }
-      currentView.triggerUpdate()
-      currentView.end()
-      currentView = newView(lifeCycle, currentLocation, ViewLoadingType.ROUTE_CHANGE)
+      onRouteChange()
+    }
+  })
+
+  // Renew view on hash changes
+  trackHash(() => {
+    if (areDifferentViews(currentLocation, location)) {
+      onRouteChange()
     }
   })
```

Comment:
What about get rid of the comments and extract all the common logic with something like:
```ts
trackHistory(renewViewOnChange)
trackHash(renewViewOnChange)

function renewViewOnChange() {
  if (areDifferentViews(currentLocation, location)) {
   currentLocation = { ...location }
   ...
  }
}
```